File: services/api/internal/export/domain/export.go
package domain

import (
	"errors"
	"time"

	"github.com/google/uuid"
)

// ExportStatus represents the lifecycle state of an export.
type ExportStatus string

const (
	StatusPending    ExportStatus = "pending"
	StatusProcessing ExportStatus = "processing"
	StatusCompleted  ExportStatus = "completed"
	StatusFailed     ExportStatus = "failed"
)

// ExportType represents the kind of export.
type ExportType string

const (
	TypePDF      ExportType = "pdf"
	TypePrintJob ExportType = "print_job"
)

var (
	ErrNotFound = errors.New("export: not found")
)

// ExportedFile represents a requested or completed file export.
type ExportedFile struct {
	ID                uuid.UUID
	TenantID          uuid.UUID
	RelatedEntityType string
	RelatedEntityID   uuid.UUID
	ExportType        ExportType
	FileKey           string
	Status            ExportStatus
	RequestedByUserID uuid.UUID
	CreatedAt         time.Time
	CompletedAt       *time.Time
	FailureReason     string
}

File: packages/api-client/src/types/export.ts
import type { UUID } from "./common"

export type ExportStatus = "pending" | "processing" | "completed" | "failed"
export type ExportType = "pdf" | "print_job"

export interface ExportedFile {
  id: UUID
  tenant_id: UUID
  related_entity_type: string
  related_entity_id: UUID
  export_type: ExportType
  file_key?: string
  status: ExportStatus
  requested_by_user_id: UUID
  created_at: string
  completed_at?: string
  failure_reason?: string
}

export interface RequestExportRequest {
  entity_type: string
  entity_id: string
}
